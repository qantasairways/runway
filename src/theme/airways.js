export const airwaysColours = {
  qantasRed: '#ee0000',
  darkRed: '#ba0000',
  focusRed: '#870000',
  transparentGrey: 'rgba(0,0,0,0.1',
  charcoal: '#323232',
  white: '#fff',
  warningOrange: '#ed710b',
  bayBlue: '#8de2e0',
  grey: '#dadada',
  darkGrey: '#626262',
  lightGrey: '#f4f5f6'
};

export const airwaysFontFamily = {
  main: '"Ciutadella", sans-serif',
  mainMedium: '"Ciutadella-Md", sans-serif'
};

export const layout = {
  borderRadius: '4px',
  gutter: '16px',
  iconSize: '32px',
  links: {
    gutter: '10px'
  }
};

export const colours = {
  primary: airwaysColours.qantasRed,
  primaryDark: airwaysColours.darkRed,
  white: airwaysColours.white,
  transparent: 'transparent',
  darkerGrey: airwaysColours.charcoal,
  darkGrey: airwaysColours.darkGrey,
  lightGrey: airwaysColours.grey,
  lighterGrey: airwaysColours.lightGrey,
  highlights: airwaysColours.bayBlue,
  warnings: airwaysColours.warningOrange
};

export const fontFamily = {
  body: airwaysFontFamily.main,
  bold: airwaysFontFamily.mainMedium,
  h1: airwaysFontFamily.main,
  h2: airwaysFontFamily.mainMedium,
  h3: airwaysFontFamily.mainMedium
};

export const fontWeight = {
  regular: 300,
  bold: 700
};

export const fontSize = {
  body: '1.125rem',
  button: '1rem',
  label: '1rem',
  large: '2.25rem'
};

export const letterSpacing = {
  small: '-0.0625rem'
};

export const buttons = {
  borderRadius: layout.borderRadius,
  clickOffset: '1px',
  width: '100%',
  height: '3.125rem'
};

export const mq = {
  tablet: '@media only screen and (min-width: 480px)'
};

export default {
  layout,
  colours,
  fontFamily,
  fontSize,
  fontWeight,
  buttons
};
