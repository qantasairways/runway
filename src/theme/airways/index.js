// TODO - remove component specific sizes
export const layout = {
  borderRadius: '4px',
  gutter: '15px',
  iconSize: '32px',
  containerMaxWidth: '1000px',
  paddingContainerMaxWidth: '1030px',
  links: {
    gutter: '10px'
  }
};

export const colours = {
  primary: '#ee0000',
  primaryDark: '#ba0000',
  white: '#fff',
  transparent: 'transparent',
  darkerGrey: '#323232',
  darkGrey: '#626262',
  mediumDarkGrey: '#888888',
  dullGrey: '#b3b3b3',
  grey: '#dadada',
  mediumGrey: '#3c3c3c',
  mediumLightGrey: '#666666',
  lightGrey: '#f4f5f6',
  lighterGrey: '#eeeeee',
  disabledGrey: '#fafafa',
  highlights: '#8de2e0',
  hightlightsLight: '#bff4f2',
  hightlightsLighter: '#e8f7f7',
  warnings: '#ed710b',
  black: '#000000'
};

export const fontFamily = {
  main: '"Ciutadella", sans-serif'
};

export const fontWeight = {
  regular: 400,
  bold: 700
};

// TODO - remove component specific sizes
export const fontSize = {
  body: '1.125rem',
  button: '1rem',
  label: '1rem',
  labelLarge: '1.375rem',
  large: '2.25rem',
  medium: '1.625rem',
  small: '0.625rem'
};

export const letterSpacing = {
  small: '-0.0625rem'
};

export const breakpoints = {
  medium: '(min-width: 768px)',
  large: '(min-width: 960px)',
  extraLarge: '(min-width: 1200px)'
};

// TODO - rename this to mediaQueries
export const mq = {
  small: '@media only screen and (min-width: 321px)',
  medium: `@media only screen and ${breakpoints.medium}`,
  large: `@media only screen and ${breakpoints.large}`,
  extraLarge: `@media only screen and ${breakpoints.extraLarge}`
};

// TODO - remove this
export const highlightInvalidField = {
  borderRadius: '2px',
  border: 'solid 2px #ed710b'
};

export default {
  layout,
  colours,
  fontFamily,
  fontSize,
  fontWeight,
  letterSpacing,
  breakpoints,
  mq,
  highlightInvalidField // this will be removed
};
