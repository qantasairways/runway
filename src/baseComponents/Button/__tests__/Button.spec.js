import React from 'react';
import { ThemedButton as Button } from '..';
import { shallow } from 'enzyme';
import theme from '../../../theme/airways';

describe('Button', () => {
  it('should render correctly with defaults', () => {
    const component = shallow(<Button theme={theme} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const component = shallow(
      <Button theme={theme} type="submit" onClick={() => 1}>
        button text
      </Button>
    );

    expect(component).toMatchSnapshot();
  });
});
