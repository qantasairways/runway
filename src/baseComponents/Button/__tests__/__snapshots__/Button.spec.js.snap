// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly with defaults 1`] = `
.emotion-0 {
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  width: 100%;
  height: 3.125rem;
  border-radius: 4px;
  border-width: 0;
  padding: 14px 24px 15px;
  font-weight: 700;
  line-height: 1.56;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  cursor: pointer;
}

.emotion-0:active {
  -webkit-transform: translate(1px,1px);
  -ms-transform: translate(1px,1px);
  transform: translate(1px,1px);
}

<button
  className="emotion-0"
  disabled={false}
  inline={false}
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button should render correctly with props 1`] = `
.emotion-0 {
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  width: 100%;
  height: 3.125rem;
  border-radius: 4px;
  border-width: 0;
  padding: 14px 24px 15px;
  font-weight: 700;
  line-height: 1.56;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  cursor: pointer;
}

.emotion-0:active {
  -webkit-transform: translate(1px,1px);
  -ms-transform: translate(1px,1px);
  transform: translate(1px,1px);
}

<button
  className="emotion-0"
  disabled={false}
  inline={false}
  onClick={[Function]}
  type="submit"
>
  button text
</button>
`;
